### ISSUE TITLE
limited client crashes in missile_update()

### ISSUE THREAD

**Issue by smcameron** [smcameron/space-nerds-in-space#166]
```
(gdb) bt
#0  0x0000000000448539 in vec3_init (vo=0x249f88, x=0.39440003, y=0.39440003, z=0.39440003) at quat.c:419
#1  0x000000000043f0a0 in update_entity_scale (e=0x249f00, scale=0.39440003) at entity.c:211
#2  0x000000000045c491 in update_missile (id=1231, timestamp=301, x=287604.04935460724, y=-61246.689437537774, z=385820.53090716735, orientation=0x7f00aeffcc10)
    at snis_limited_client.c:1423
#3  0x0000000000468a44 in process_update_missile_packet () at snis_limited_client.c:4962
#4  0x000000000046e78a in gameserver_reader (arg=<error reading variable: Cannot access memory at address 0x7f003f4ccca5>) at snis_limited_client.c:7185
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
(gdb) up
#1  0x000000000043f0a0 in update_entity_scale (e=0x249f00, scale=0.39440003) at entity.c:211
211		vec3_init(&e->e_scale, scale, scale, scale);
(gdb) up
#2  0x000000000045c491 in update_missile (id=1231, timestamp=301, x=287604.04935460724, y=-61246.689437537774, z=385820.53090716735, orientation=0x7f00aeffcc10)
    at snis_limited_client.c:1423
1423						update_entity_scale(t, (0.5 * THRUST_FLARE_SCALE +
(gdb) print t
$1 = (struct entity *) 0x249f00
(gdb) print *t
Cannot access memory at address 0x249f00
(gdb) print go[i].tsd.missile
$2 = {target_id = 4294967295, origin = 0, thrust_entity = {0x7f00aeffcc00, 0x249f00}}
(gdb)
```

Something to do with thrust flare entities... the entity pointer is not valid.

**Comment by smcameron** [comment]
I noticed that this code was getting triggered in add_entity()

```
  printf("Out of entities at %s:%d\n", __FILE__, __LINE__);
```

Increasing the allowable number of entities seems to make the problem go away:

snis_client.c:

```
@@ -21207,7 +21210,7 @@ int main(int argc, char *argv[])
        setup_physical_io_socket();
        setup_natural_language_fifo();
        setup_text_to_speech_thread();
-       ecx = entity_context_new(5000, 5000);
+       ecx = entity_context_new(10000, 10000);
 
        ssgl_register_for_bcast_packets(lobby_list_change_notification); /* must happen after init_net_setup_ui() */
        snis_protocol_debugging(1);
```

This solution seems not very satisfactory though.  It should withstand entity allocation failures.

**Comment by smcameron** [comment]
Ok, I figured out why the crash happened.  If there is an entity allocation failure for the parent entity (e.g. the missile entity) o->tsd.missile.thrust_entity[] wasn't get initialized.  Fixed by f564585f27c14e092c94881db664642bb5dadf12

I don't know why the limited client is getting entity allocation failures but the opengl client isn't though.
Seems like they both should, or neither should, they should not be different in this regard.
